docs-website/CONTRIBUTING.md

```diff
@@ -127,87 +184,321 @@ Data loss or security-impacting issues.
+- [ ] Navigation updated (`sidebars.js` or `reference-sidebars.js`)
+- [ ] Internal links verified (no broken anchors)
+- [ ] Code samples tested and include language tags
+- [ ] Images optimized and include alt text
+- [ ] Local build passes (`npm run build`)
+- [ ] Vercel preview deployment succeeds (fix any deployment errors)
+- [ ] Vale checks pass or issues are addressed
+- [ ] Conventional commit message format used in PR title
+- [ ] PR description includes context and related issues
+
+**PR title format:**
+
+Use conventional commits in the PR title:
+
+```
+docs: add troubleshooting guide for pipelines
+docs: fix typo in installation instructions
+docs: update API reference links
+```
 
-- Keep existing URLs stable whenever possible. If you move a page, retain its `slug` so the path does not change.
-- Update `sidebars.js` or `reference-sidebars.js` accordingly.
-- If a URL must change, coordinate with maintainers to set up a redirect strategy and avoid broken inbound links.
+**PR description:**
 
-## Images and assets
+Include:
+- Summary of changes
+- Screenshots (if UI changes are visible)
+- Related issues (for example, "Fixes #123")
+- Testing performed
+- Notes for reviewers
 
-- Place shared images in `static/img/` or per-section assets in `docs/assets/`.
-- Use descriptive filenames and alt text.
-- Prefer optimized images; the site uses responsive image tooling to reduce payload size.
+## Review Process
 
-## API Reference contributions
+1. Open a PR from your branch to `main`
+2. Automated checks will run (Vale linting, build validation)
+3. Maintainers will review your changes
+4. Address any requested changes
+5. Once approved and checks pass, a maintainer will merge
+6. Your changes will be deployed automatically
 
-- The API documentation is automatically generated from Python docstrings in the Haystack codebase. To change API docs, edit the relevant docstrings in the [Haystack repository](https://github.com/deepset-ai/haystack) and open a PR there.
-- Do not edit files under `reference/` in this repository. They are regenerated and any manual changes will be overwritten.
+## Accessibility and Inclusivity
 
-## Accessibility and inclusivity checklist
+Ensure your documentation is accessible to all users:
 
-- Alt text for images
-- Descriptive link text
-- Clear, concise sentences; avoid jargon when possible
-- Inclusive language and examples
+- **Alt text:** Provide descriptive alt text for all images
+- **Link text:** Use descriptive link text (not "click here")
+- **Language:** Use clear, concise sentences; avoid jargon where possible
+- **Examples:** Use inclusive language and diverse examples
+- **Headings:** Use proper heading hierarchy (don't skip levels)
+- **Code blocks:** Include language tags for proper syntax highlighting
 
-## Pull request checklist
+## Getting Help
 
-Before submitting your PR:
+**Questions about contributing:**
+- Review this guide and the [README](./README.md)
```

Comment:
maybe we should mention https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md